python_bindings/rig_scp.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <Python.h>
+#include <rs.h>
+
+/* SCPConnection type. */
+typedef struct {
+  PyObject_HEAD
+  rs_conn_t *connection;
+} SCPConnection;
+
+static PyObject* SCPConnection_new(PyTypeObject *type, PyObject *args,
+                                   PyObject *kwds)
+{
+  // Create a new SCPConnection
+  SCPConnection *self;
+  self = (SCPConnection *)type->tp_alloc(type, 0);
+  if(self != NULL)
+  {
+    self->connection = NULL;
+  }
+
+  return (PyObject *) self;
+}
+
+static int SCPConnection_init(SCPConnection *self, PyObject *args,
+                              PyObject *kwds)
+{
+  // Parse the arguments and keyword arguments
+  const char *hostname = NULL;  // Host to communicate with
+  unsigned int port = 17893;  // Port to communicate on
+  double timeout = 0.5;  // Duration to timeout after
+  unsigned int n_tries = 5;  // Number of retries before transmitting
+  unsigned int window_size = 8;  // Number of allowable outstanding packets
+
+  static char *kwlist[] = {"hostname", "port", "timeout", "n_tries",
+                           "window_size", NULL};
+
+  if(!PyArg_ParseTupleAndKeywords(args, kwds, "s|IdII", kwlist, &hostname,
+                                  &port, &timeout, &n_tries, &window_size))
+    return -1;
+
+  // Convert the port integer to a string
+  char port_str[5];  // 16 bits -> 5 characters max (65536)
+  sprintf(port_str, "%5d", port);
```

Comment:
For certainty, I'd mask port with 0xFFFF otherwise -ve or over-large values will cause a buffer overrun. (Marginally less faff than snprintf.